x-pack/plugins/discover_log_explorer/public/components/dataset_selector/sub_components/datasets_popover.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiButton,
+  EuiHorizontalRule,
+  EuiPanel,
+  EuiPopover,
+  EuiPopoverProps,
+  EuiTitle,
+  useIsWithinBreakpoints,
+} from '@elastic/eui';
+import styled from '@emotion/styled';
+import { DATA_VIEW_POPOVER_CONTENT_WIDTH, POPOVER_ID, selectDatasetLabel } from '../constants';
+import { getPopoverButtonStyles } from '../utils';
+
+const panelStyle = { width: DATA_VIEW_POPOVER_CONTENT_WIDTH };
+interface DatasetsPopoverProps extends Omit<EuiPopoverProps, 'button'> {
+  children: React.ReactNode;
+  onClick: () => void;
+  title: string;
+}
+
+export const DatasetsPopover = ({ children, onClick, title, ...props }: DatasetsPopoverProps) => {
+  const isMobile = useIsWithinBreakpoints(['xs', 's']);
+
+  const buttonStyles = getPopoverButtonStyles({ fullWidth: isMobile });
+
+  return (
+    <EuiPopover
+      id={POPOVER_ID}
+      button={
+        <EuiButton
+          css={buttonStyles}
+          iconType="arrowDown"
```

Comment:
Too bad the drop-down arrow takes up the icon slot in the button. It would have been great to show the icon of the currently selected integration in front of the title. @formgeist would it be an option to drop the arrow in favor of the integration icon?

Reply:
Just FYI, in the current dataView picker of discover a custom icon is added in front of the title whenever it is an ad-hoc data view, so we could do something similar on a technical level. Just wondering if it will result in too many icons in the button...

Reply:
I think the visibility of the integration logo is quite important to immediately recognize the selected dataset. If we can make it part of the title itself and leave the dropdown arrow I think we have a good solution for both indicating the selected dataset and indicating the interaction once the button is clicked.

Reply:
Sounds good to me @formgeist, and to think in the long term, what would be an icon when we'll have multi-selection? Do we want to avoid the icon in case more than a Dataset is selected?

Reply:
Good point! In relation to this I'd propose that we combine the selected datasets into a "[3] selected" summarized button. I think listing out the integrations comma-separated isn't a great experience. When the user would hover the summarized button we could show them a list of the selected datasets. WDYT? It's not something we need until we have multi-selection of course.

Reply:
>  I'd propose that we combine the selected datasets into a "[3] selected"
Agree, we were already looking into something like that and Felix proposed a design really similar.
Sounds good then, I'll look at implementing the integration icon for single-selection dataset and hide it for multi-selection.

Reply:
@formgeist @weltenwort does this looks as expected?

https://github.com/elastic/kibana/assets/34506779/b1eb3bab-a7f1-45b7-a1bf-1309dcf3b89d

Reply:
@tonyghiani with the icon there, yes, but now the titles look superfluous with their `[Kubernetes] Kubernetes....` prefixes 🤔 I also would want to remove them in case users customize the prefixes to their liking. Are those titles realistic when it comes to the dataset names?

Reply:
Those titles are taken straight from the package. Unfortunately there's no guarantee that the dataset title will also include the integration name. But maybe it would still be sufficient to only show the icon and not the integration name.

Reply:
Ok I'll remove the `[<package-name>]` from the title 👌